dataframely/columns/datetime.py

```diff
@@ -329,16 +341,26 @@ def validation_rules(self, expr: pl.Expr) -> dict[str, pl.Expr]:
         return result
 
     def sqlalchemy_dtype(self, dialect: sa.Dialect) -> sa_TypeEngine:
+        timezone_enabled = self.time_zone is not None
         match dialect.name:
             case "mssql":
                 # sa.DateTime wrongly maps to DATETIME
-                return sa_mssql.DATETIME2(6)
+                return sa_mssql.DATETIME2(6, timezone=timezone_enabled)
+            case "postgresql":
+                return sa_postgresql.TIMESTAMP(timezone=timezone_enabled)
```

Comment:
Do you need this extra condition for postgres or does `sa.DateTime(timezone=timezone_enabled)` map to `TIMESTAMPTZ` anyways? We're currently not maintaining 100% coverage (which we should) because we're not using the default case.